PurchasesTests/TestHelpers/ObjCThrowExceptionMatcher.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ObjCThrowExceptionMatcher.swift
+//  PurchasesTests
+//
+//  Created by Joshua Liebowitz on 6/10/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import Nimble
+
+enum ObjCException: String {
+    
+    case parameterAssert = "NSInternalInconsistencyException"
+    
+}
+
+func expectToThrowException(_ named: ObjCException? = nil, closure: @escaping  () -> Void) -> Void {
```

Comment:
❤️

Reply:
out of curiosity, was there a reason to go with `void` and perform the expect + the matching here instead of returning a predicate?

Reply:
I started by attempting to a real matcher that Nimble could use (which would return a predicate), but it got confusing pretty quickly. I didn't want to spend much time figuring out how to make it work nor really dive into how Nimble worked either. I figured, for now, this is good enough and I could totally revisit it later if we decided it would be helpful. Basically, while the best solution would be to write a true matcher, IMO this is GoodEnough™️